go/libkb/globals.go

```diff
@@ -226,7 +228,7 @@ func (g *GlobalContext) createLoginStateLocked() {
 		g.loginState.Shutdown()
 	}
 	g.loginState = NewLoginState(g)
-	g.ActiveDevice = new(ActiveDevice)
+	g.ActiveDevice.clear(g.loginState.account)
```

Comment:
could you change this to something like

    g.loginState.Account(func (a *Account) {
        g.ActiveDevice.clear(a)
    }, "ActiveDevice.clear")

as an extra protection?  Would prefer not to start accessing anything private in loginState.

thanks!